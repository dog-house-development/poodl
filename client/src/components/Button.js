import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    onClick: PropTypes.func.isRequired,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string.isRequired
};

const defaultProps = {
    size: 'medium',
    type: 'primary'
};

class Button extends Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(e) {
        e.preventDefault();
        this.props.onClick();
    }

    render() {
        return (
            <button onClick={this.handleClick} className={`button ${this.props.size} ${this.props.type}`}>
                {this.props.content}
            </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
