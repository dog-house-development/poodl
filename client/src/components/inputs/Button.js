import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    onClick: PropTypes.func,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    kind: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string.isRequired,
    type: PropTypes.string
};

const defaultProps = {
    size: 'medium',
    kind: 'primary'
};

class Button extends Component {
    // constructor(props) {
    //     super(props);
    // }

    render() {
        return (
            <button
                onClick={this.props.onClick}
                className={`button ${this.props.size} ${this.props.kind}`}
                type={this.props.type} >
                {this.props.content}
            </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
