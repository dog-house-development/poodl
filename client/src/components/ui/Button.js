import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    onClick: PropTypes.func,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    kind: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string.isRequired,
    type: PropTypes.string,
    formButton: PropTypes.bool,
    dropdownButton: PropTypes.bool
};

const defaultProps = {
    size: 'medium',
    kind: 'primary',
    dropdownButton: false
};

class Button extends Component {
    render() {
        return (
            <button
                onClick={this.props.onClick}
                className={`button ${this.props.size} ${this.props.kind} ${this.props.formButton ? 'form-button' : ''}`}
                type={this.props.type}
            >
                {this.props.content}
            </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
