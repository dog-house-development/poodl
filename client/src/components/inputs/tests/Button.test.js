import React from "react";
import ReactDOM from "react-dom";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import _ from "lodash";

import Button from "../Button";

configure({ adapter: new Adapter() });

let state, props, wrapper, instance;
beforeEach(() => {
  props = {
    onClick: () => {},
    size: "medium",
    kind: "primary",
    content: "Press me",
    type: "normal"
  };
  wrapper = shallow(<Button {..._.assign({}, state, props)} />);
  instance = wrapper.instance();
});

describe("render", () => {
  it("should render correctly", () => {
    expect(instance).toMatchSnapshot();
  });
});
