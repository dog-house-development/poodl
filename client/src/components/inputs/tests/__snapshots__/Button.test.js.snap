// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render correctly 1`] = `
Button {
  "context": Object {},
  "props": Object {
    "content": "Press me",
    "kind": "primary",
    "onClick": [Function],
    "size": "medium",
    "type": "normal",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <Button
        content="Press me"
        kind="primary"
        onClick={[Function]}
        size="medium"
        type="normal"
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <button
        className="button medium primary"
        onClick={[Function]}
        type="normal"
      >
        Press me
      </button>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;
